### PR TITLE
Feature/presolve domain propagation

### PR DESCRIPTION
**Pull request by elham-azd**

### CODE REVIEW

File: dwave/preprocessing/include/dwave/presolve.h [hunk 1]
```diff
@@ -511,7 +621,9 @@ void Presolver<bias_type, index_type, assignment_type>::apply() {
         changes |= technique_check_for_nan();
         // *-- remove single variable constraints
         changes |= technique_remove_single_variable_constraints();
-        // *-- tighten bounds based on vartype
+        // *-- domain propagation
```
Comment (elham-azd) [inline dwave/preprocessing/include/dwave/presolve.h:624]: Is this the right order or we want to do the domain propagation after tightening the bounds?

Comment (alexzucca90) [reply dwave/preprocessing/include/dwave/presolve.h:624]: I think I would first tighten the bounds and then domain propagate. At the beginning of paragraph 3.2 of the paper you followed (bottom of page 7), they assume that the bounds are already tightened, so it makes sense to do that first.

Comment (elham-azd) [reply dwave/preprocessing/include/dwave/presolve.h:626]: Right! Thanks Alex. Done.

**Comment by arcondello** [comment]
Going to merge this into a dev branch and make some additional changes from there.

**Pull request merged**